### PR TITLE
refactor(DateTimeUtils)

### PR DESCRIPTION
**Pull request by kaligrafy**
Replace string parsing with Intl.DateTimeFormat.formatToParts() in getTimeZoneOffsetMinutes() for more reliable timezone offset calculation. This eliminates locale-specific format dependencies and improves robustness.

- Use formatToParts() API instead of string parsing
- Improve function documentation with clearer behavior descriptions
- Add explicit sign convention documentation
- Enhance test assertions for exact value verification
- Fix fractional seconds test to verify truncation behavior
- project config: change format of start/end date to be ISO date time string with timezone offset

## Summary by CodeRabbit

* **New Features**
  * Project configuration now exposes optional start/end datetimes with timezone offsets and sets a default country code (CA).
  * New date/time utilities for ISO-with-timezone formatting, epoch conversion, and timezone-offset handling.

* **Tests**
  * Updated project config tests to cover the new datetime fields.
  * Extensive new tests validating conversions, formatting, timezone offsets, DST edge cases, and round-trip scenarios.

### CODE REVIEW

File: packages/evolution-common/src/config/project.config.ts [hunk 1]
```diff
@@ -25,6 +26,27 @@ export type EvolutionProjectConfiguration = {
     /** Used for Google Maps localization. See
      * https://developers.google.com/maps/coverage for possible region codes */
     region: string;
+    /**
+     * Two-letter country code for the survey. Used for phone number validation.
+     * Defaults to 'CA'.
+     * */
```
[the remark this answers is not in the document]
Comment (kaligrafy) [reply packages/evolution-common/src/config/project.config.ts]: removed from commit

File: packages/evolution-common/src/config/project.config.ts [hunk 2]
```diff
@@ -143,64 +160,67 @@ export type EvolutionProjectConfiguration = {
...
+    endDateTimeWithTimezoneOffset: undefined,
+    hideStartButtonOnHomePage: false,
+    introductionTwoParagraph: false,
+    introBanner: false,
+    bannerPaths: {},
+    introLogoAfterStartButton: false,
+    logoPaths: {},
+    detectLanguageFromUrl: true,
+    detectLanguage: false,
+    languageNames: { en: 'English', fr: 'Français' },
```
[the remark this answers is not in the document]
Comment (kaligrafy) [reply packages/evolution-common/src/config/project.config.ts]: fixed

File: packages/evolution-common/src/config/project.config.ts [hunk 3]
```diff
@@ -25,6 +26,24 @@ export type EvolutionProjectConfiguration = {
     /** Used for Google Maps localization. See
      * https://developers.google.com/maps/coverage for possible region codes */
     region: string;
+    /**
+     * Start date time for the survey in ISO date time string format with timezone offset
+     * (YYYY-MM-DDTHH:MM:SS-/+HH:MM). Example: 2025-01-01T00:00:00-05:00.
+     * Interviews started before this date and time should be invalidated and/or ignored.
+     * If both startDateTimeWithTimezoneOffset and endDateTimeWithTimezoneOffset are defined,
+     * endDateTimeWithTimezoneOffset must be after startDateTimeWithTimezoneOffset.
+     * Provide the timezone offset so we can calculate the correct unix epoch.
```
Comment (tahini) [inline packages/evolution-common/src/config/project.config.ts]: What if it is not provided. Say someone just enters '2025-01-01' because they don't care, will it still work? Would any string convertible to datetime format work?

File: packages/evolution-common/src/utils/DateTimeUtils.ts [hunk 4]
```diff
@@ -0,0 +1,135 @@
...
+                getValue('month') - 1,
+                getValue('day'),
+                getValue('hour'),
+                getValue('minute'),
+                getValue('second')
+            );
+        };
+
+        // Get UTC timestamp
+        const utcFormatter = new Intl.DateTimeFormat('en-CA', { ...formatOptions, timeZone: 'UTC' });
```
Comment (tahini) [inline packages/evolution-common/src/utils/DateTimeUtils.ts]: Why hardcode en-CA here?

File: packages/evolution-common/src/utils/__tests__/DateTimeUtils.test.ts [hunk 5]
```diff
@@ -0,0 +1,479 @@
...
+            const epochMilliseconds = -86400000; // 1969-12-31
+            const result = epochToDate(epochMilliseconds / 1000);
+            expect(result.toISOString()).toBe('1969-12-31T00:00:00.000Z');
+        });
+
+        test('should handle fractional seconds by truncating', () => {
+            const epochMilliseconds = 1609459200000.999;
+            const result = epochToDate(epochMilliseconds / 1000);
+            // Fractional seconds should be truncated to the nearest millisecond
+            const expectedTimestamp = 1609459200000; // '2021-01-01T00:00:00.000Z'
```
Comment (tahini) [inline packages/evolution-common/src/utils/__tests__/DateTimeUtils.test.ts]: peut-être faire un test avec des ms non zéro aussi?

File: packages/evolution-common/src/config/project.config.ts [hunk 6]
```diff
@@ -143,64 +162,67 @@ export type EvolutionProjectConfiguration = {
...
+    bannerPaths: {},
+    introLogoAfterStartButton: false,
+    logoPaths: {},
+    detectLanguageFromUrl: true,
+    detectLanguage: false,
+    languageNames: { en: 'English', fr: 'Français' },
+    title: { en: 'Survey', fr: 'Enquête' },
+    postalCodeRegion: 'other',
+    personColorsPalette: [
+        // FIXME See this issue https://github.com/chairemobilite/evolution/issues/1246
```
Comment (samuel-duhaime) [inline packages/evolution-common/src/config/project.config.ts]: Write one sentence for the issue.

File: packages/evolution-common/src/config/project.config.ts (hunk 3 above)
Comment (kaligrafy) [reply packages/evolution-common/src/config/project.config.ts]: On doit vérifier ça dans le survey... Comment tester ça dans evolution?

File: packages/evolution-common/src/utils/DateTimeUtils.ts [hunk 7]
```diff
@@ -0,0 +1,135 @@
...
+): string => {
+    return new Intl.DateTimeFormat('sv-SE', {
+        dateStyle: 'short',
+        timeStyle: withTime ? 'medium' : undefined,
+        timeZone
+    }).format(date);
+};
+
+/**
+ * Get the time zone offset in minutes for a given date and time zone.
```
[the remark this answers is not in the document]
Comment (tahini) [reply packages/evolution-common/src/utils/DateTimeUtils.ts]: > I worked a lot to make this function work. It is not used right now, but we will need this to offset epochs to get the datetime in any timezone. Do you want me to remove it altogether?

I'd say yes. Why work on something we have no current need for? And when the need comes, we may find another simpler solution with an actual use case. Getting the date time in any timezone is just a conversion, no?

File: packages/evolution-common/src/utils/DateTimeUtils.ts (hunk 4 above)
Comment (tahini) [reply packages/evolution-common/src/utils/DateTimeUtils.ts]: > This was suggested by both codeRabbit and Claude... It seems en-CA is one of the nearest to ISO-8601 standard

In a jsdoc earlier, your said sv-SE is actually the ISO8601
